parcels/kernel.py

```diff
@@ -378,7 +378,7 @@ def execute(self, pset, endtime, dt, recovery=None, output_file=None, execute_on
 
         def remove_deleted(pset):
             """Utility to remove all particles that signalled deletion"""
-            indices = pset.particle_data['state'] == OperationCode.Delete
+            indices = pset.collection._data['state'] == OperationCode.Delete
```

Comment:
also here, better have a setter/getter for the data array, parametrized by the name of the variable.

Reply:
This line has already been replaced with an alternative, that does not use the _data array directly, by Raoul.